src/System.Private.ServiceModel/tools/CertificateGenerator/CertificateManager.cs

```diff
@@ -186,6 +199,16 @@ public static X509Certificate2 CreateAndInstallLocalMachineCertificates(Certific
                 InstallCertificateToRootStore(rootCertificate);
                 InstallCertificateToMyStore(hostCert, certificateCreationSettings.ValidityType == CertificateValidityType.Valid);
                 s_localCertificate = hostCert;
+
+                // Create the PeerTrust cert
+                certificateCreationSettings = new CertificateCreationSettings()
+                {
+                    FriendlyName = "WCF Bridge - UserPeerTrustCertificateResource",
+                    Subject = fqdn,
+                    SubjectAlternativeNames = new string[] { fqdn, hostname, "localhost" }
+                };
+                var peerCert = certificateGenerator.CreateMachineCertificate(certificateCreationSettings).Certificate;
+                InstallCertificateToTrustedPeopleStore(peerCert, certificateCreationSettings.ValidityType == CertificateValidityType.Valid);
             }
```

Comment:
Well here is the problem -- let's brainstorm what we want: The intent is that we want at least one cert in TrustedPeople so we can exercise the PeerTrust-only path.  I chose to make it chain-trusted as well so that we could test the ChainTrust path. Therefore the cert passes validation for PeerTrust, PeerOrChainTrust, and ChainTrust (and gives us coverage of all the positive code paths in the validator check). But once we get past the X509CertificateValidator checks, the cert still has to work end to end.  I could not find a way to make a cert work all the way through TCP with any less than what is here.  When I used something closer to the client cert, I was blocked by EndpointIdentity checks (and we've disabled many of the alternate EndpointIdentity code paths).  So the issue is this -- we need a cert that exercises X509CertificateValidation checks *and* is trusted by all the other layers in the stack.

How do you feel about letting this PR merge (after addressing other concerns) and then come back with a 2nd PR for just the cert if we can find a way to make that happen?  I've had to rebase this 3 times already.

---
In reply to: [71627124](https://github.com/dotnet/wcf/pull/1382#discussion_r71627124) [](ancestors = 71627124)